[MRELEASE-977] makes the release:branch goal ask for branch name if not given

hgschmie opened the pull request:
Make branchName in the BranchReleaseMojo no longer required.
Adds the InputVariablesPhase to the list of branch phases; change the prompt slightly
when doing a branch name input.

hgschmie commented:
Hello @asfgit, this is for MRELEASE-977.

Tunaki commented on maven-release-manager/src/main/java/org/apache/maven/shared/release/phase/InputVariablesPhase.java:
```diff
@@ -159,9 +167,18 @@ public ReleaseResult execute( ReleaseDescriptor releaseDescriptor, ReleaseEnviro
+                    {
+                        tag =
+                            prompter.prompt( "What is the branch name for \"" + project.getName() + "\"? ("
+                                             + project.getGroupId() + ":" + project.getArtifactId() + ")", defaultTag );
```
Instead of defaulting to the calculated tag name when prompting for the branch name, it would be better to fail if the user inputs an empty line. The default will almost never be the right one, and it would align with the current behaviour: for example, in batch mode without a branch name, the plugin should fail.

Additionally, this could be refactored in a method `getScmReleaseLabel`, returning the release label to use based on the default tag and release descriptor, for clarity.

hgschmie replied:
ok, will update accordingly.

Tunaki commented:
Thanks for the new commit. There is still the case where the release is not in interactive mode that needs to be handled. The user shouldn't be able to pass an empty branch name. Could you add this and squash the commits?

hgschmie commented:
added an explicit check for empty branch and a test; rebased and squashed. 

It now behaves exactly like the current 2.5.3 release:branch mojo when running in non-interactive mode.

Tunaki commented:
Thanks! Merged in [r1779510](http://svn.apache.org/viewvc?rev=1779510&view=rev). Could you close this PR?

hgschmie commented:
merged. Thanks!
